src/System.IO.Pipes/src/System.IO.Pipes.csproj

```diff
@@ -254,6 +254,12 @@
     <Reference Include="System.Threading.Overlapped" />
     <Reference Include="System.Threading.Tasks" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
+    <Reference Include="System.IO.Pipes.AccessControl" />
+    <Reference Include="System.Security.AccessControl" />
+    <Reference Include="System.Security.Principal.Windows" />
+    <Reference Include="System.Security.Claims" />
+  </ItemGroup>
   <ItemGroup Condition="'$(TargetsUnix)' == 'true'">
```

Comment:
What are we going to do on unix?

Reply:
I'm still reading about Unix security. Not yet what dependencies we're going to need. If any of the ones imported for windows is needed I will move it to the common ItemGroup.

Reply:
On Unix we will probably want to create a directory that's chmod'd to only be accessible by the current user, and then create the domain socket in that directory.

Reply:
That is the main thing still missing: right now there is enforcement on the client but nothing on the server side.

Reply:
Yes, I'm working on it.

Reply:
You may want to be careful about Mac if you plan to use the temp directory for this. The max path length for a unix domain socket on Mac is ~104 characters. Mac generates per-user temp paths that can be very large, and the standard prefixing often adds up to ~50 characters already. If the generated folder name is too long you could end up leaving the user with almost no space for their pipe name.

Reply:
> You may want to be careful about Mac if you plan to use the temp directory for this. The max path length for a unix domain socket on Mac is ~104 characters. Mac generates per-user temp paths that can be very large, and the standard prefixing often adds up to ~50 characters already. If the generated folder name is too long you could end up leaving the user with almost no space for their pipe name.

I'm not changing the way logic how we create the socket, the plan and what I'm doing is to just do a chmod 600 to give write and read access to the owner only. So I will not change the behavior and I guess it works fine since it's been like for some time already.

Reply:
> I'm not changing the way logic how we create the socket, the plan and what I'm doing is to just do a chmod 600

On what?  chmod'ing the domain socket path won't work.  And I believe @agocke's comment is that by adding in the directory, we'll be consuming some of the space available for the full name.

But I don't know of a better way.

Reply:
I see what you're saying. What if for the UserOnly pipes instead of adding the corefx prefix in the pipe name we use that space that we're already taking, for the directory name? 

https://github.com/dotnet/corefx/blob/40cbd5636851093309b841db4d53af369b7ef1c6/src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs#L29

Reply:
We could create a directory that contains the userid so that it is unique and related to this user and we don't generate random names, so maybe something like: `tempdir/{euid}_fx/{pipename}` where an euid is using the 16 bits. As the docs say:

>The original Linux getuid() and geteuid() system calls supported only 16-bit user IDs. Subsequently, Linux 2.4 added getuid32() and geteuid32(), supporting 32-bit IDs. The glibc getuid() and geteuid() wrapper functions transparently deal with the variations across kernel versions.

Which at the end will be using less space than what `CoreFxPipe_` prefix is already using on a regular named pipe.

@stephentoub thoughts?

Reply:
@stephentoub noting for posterity that on Roslyn they currently don't use this approach, they connect blindly then check GetPeerID() against GetEUid().

https://github.com/dotnet/roslyn/blob/c7a465fabaf688affbd5897d7348431eeb2059b5/src/Compilers/Shared/BuildServerConnection.cs#L490

Reply:
@stephentoub any reason to prefer one over the other? It seems like getpeerid() is for this exact purpose. https://linux.die.net/man/3/getpeereid

Reply:
@danmosemsft I am starting to look back at this to add the tests this week, and I think your point about `getpeerid` is valid. The only thing that cross my mind is that the existing API on the server side currently has to either throw an exception for each mismatched peer (after the socket connection) or simply ignore them - neither seems good (perhaps a new API in the future?). In this sense the directory is helpful because prevents connections before it reaches the server socket. OTOH it will introduce some small difference between Unix and Windows in at least some (mis)configurations: 


0. Assume same user for all steps;
1. Create server *without* current user only flag;
2. Create client *with* current user only flag;

On Windows: client will connect fine;

On Unix: client, depending on API, hangs or throws timeout exception (since the path for the named pipes is different between server and client) 


All of this to say that let's keep the directory, but also add the `getpeerid` check also on the server side (extra precaution), it is already done on the client side.